OpenAI.SDK/Extensions/HttpclientExtensions.cs

```diff
@@ -28,6 +30,8 @@ public static HttpResponseMessage PostAsStreamAsync(this HttpClient client, stri
         using var request = new HttpRequestMessage(HttpMethod.Post, uri);
         request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/event-stream"));
         request.Content = content;
+        var jsonString = JsonSerializer.Serialize(requestModel,settings);
+        request.Content.Headers.ContentLength = Encoding.UTF8.GetByteCount(jsonString);
```

Comment:
the content here has yet another header and there the length doesn't get set automatically for whatever reason.